src/lsm/scan_tree.zig

```diff
@@ -271,6 +288,73 @@ pub fn ScanTreeType(
                 },
                 .needs_data => return error.ReadAgain,
                 .buffering => unreachable,
+                .aborted => return null,
+            }
+        }
+
+        /// Probing a scan modifies the key_min/key_max range, either moving the scan if the next
+        /// key is already buffered, or causing it to read again from the new key range.
+        pub fn probe(self: *ScanTree, key: Key) void {
```

Comment:
Clarify that in both cases the outcome of `probe()` is that the next key from the `ScanTree` will be either probe-key, or (if probe-key doesn't exist in the tree) the next key after probe-key in the index.